lib/queue.js

```diff
@@ -551,6 +551,13 @@ Queue.prototype.processStalledJob = function(job){
           if(!isMember){
             return _this.processJob(job);
           }
+
+          return job.releaseLock(_this.token);
+        }).catch(function(err) {
```

Comment:
`finally` would not be ideal in this case because `finally` runs regardless of the fate of the promise, that is, even if the `then` function evaluates and handles the lock release, `finally` will attempt to re-release the lock, which in this case, could release the lock of other resource that took it. We need to ensure that `releaseLock` is called exactly once in this block, with `finally` we'll be in a situation where releaseLock will be called at least once instead.

Regarding disposers, I totally like the idea of it, that's how resources are disposed reliably on Go or C# for example, but again it seems to me we'll end up in a situation where releaseLock will be called more than once